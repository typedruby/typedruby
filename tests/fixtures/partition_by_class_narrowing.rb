# @typedruby

module Enumerable::[EnumType]
  include Kernel
end

def test1(Enumerable::[[String, Integer]] enum) => nil
  if enum.is_a?(Hash)
    reveal_type(enum.keys.first)
  end

  nil
end
