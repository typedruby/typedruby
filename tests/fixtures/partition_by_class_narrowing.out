error: Revealed type is: NilClass | String

        @ __ROOT__/tests/fixtures/partition_by_class_narrowing.rb:9
      9 |      reveal_type(enum.keys.first)
                           ^^^^^^^^^^^^^^^ expression
